### PR TITLE
PHP 8.2 updates

### PR DESCRIPTION
**Pull request by cviebrock**
Various fixes for PHP 8.2
- Code style
- Rector
- phpstan
- refactor some functions that were removed in PHP 8.0 (in test suite; so never affected our projects)

### CODE REVIEW

File: MDB2/Driver/Common.php [hunk 1]
```diff
@@ -480,23 +492,25 @@ public function errorInfo($error = null)
      * @param bool   $dummy2   not used
      *
      * @return PEAR_Error instance of a PEAR Error object
+     *
      * @see    PEAR_Error
      */
-    public function raiseError($code = null,
+    public function raiseError(
```
Comment (gauthierm) [inline MDB2/Driver/Common.php:498]: I think this is supposed to be `static public`.

File: MDB2/Driver/Manager/Common.php [hunk 2]
```diff
@@ -492,73 +466,9 @@ public function vacuum($table = null, $options = array())
      *                                 should be set to another associative array with the properties
      *                                 of the fields to be added. The properties of the fields should
      *                                 be the same as defined by the MDB2 parser.
-     *
```
Comment (gauthierm) [inline MDB2/Driver/Manager/Common.php:495]: These docs should not get deleted.

File: MDB2/Result/Common.php [hunk 3]
```diff
@@ -389,19 +376,18 @@ public function nextResult()
...
      * Retrieve the names of columns returned by the DBMS in a query result or
      * from the cache.
      *
-     * @param   bool    If set to true the values are the column names,
-     *                  otherwise the names of the columns are the keys.
-     * @return  mixed   Array variable that holds the names of columns or an
-     *                  MDB2 error on failure.
-     *                  Some DBMS may not return any columns when the result set
-     *                  does not contain any rows.
+     * @param   bool    if set to true the values are the column names,
```
Comment (gauthierm) [inline MDB2/Result/Common.php]: should be `@param bool $flip`

File: MDB2/Driver/Manager/Common.php (hunk 2 above)
Comment (cviebrock) [reply MDB2/Driver/Manager/Common.php:495]: Restored (and reformatted).

**Pull request merged**